game-core/src/system/ally.rs

```diff
@@ -190,13 +194,21 @@ impl<'s> System<'s> for Spawner {
                     current_frame: 0,
                 };
 
+                let max_hp = config.ally.max_hp;
+
+                let character = Character {
+                    hp: max_hp,
+                    max_hp: max_hp,
```

Comment:
this will probably cause a lint, when the field matches the binding name you can write it as `max_hp,` rather than `max_hp: max_hp,`